app/src/main/kotlin/com/rockspin/bargainbits/ui/watch_list/WatchListActivity.kt

```diff
@@ -0,0 +1,103 @@
+package com.rockspin.bargainbits.ui.watch_list
+
+import android.os.Bundle
+import android.support.v7.widget.Toolbar
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.widget.Button
+import butterknife.Bind
+import butterknife.ButterKnife
+import com.jakewharton.rxbinding2.view.RxView
+import com.rockspin.bargainbits.R
+import com.rockspin.bargainbits.ui.BaseMvpActivity
+import com.rockspin.bargainbits.ui.views.deallist.DealsListView
+import com.rockspin.bargainbits.ui.views.deallist.view.DealsListPresenter
+import com.rockspin.bargainbits.ui.views.deallist.view.DealsListViewImpl
+import hu.akarnokd.rxjava.interop.RxJavaInterop
+import io.reactivex.Observable
+import javax.inject.Inject
+
+class WatchListActivity : BaseMvpActivity<WatchListView, WatchListPresenter>(), WatchListView, DealsListViewImpl.DealsListContainer {
+
+    @Inject override lateinit var presenter: WatchListPresenter
+    @Inject lateinit var dealsListPresenter: DealsListPresenter
+
+    @Bind(R.id.watchlist_toolbar) lateinit var toolbar: Toolbar
+    @Bind(R.id.watchlist_deals_list) lateinit var dealsList: DealsListView
+    @Bind(R.id.watchlist_layout_loading) lateinit var loadingView: View
+    @Bind(R.id.watchlist_button_back) lateinit var backButton: Button
+    @Bind(R.id.watchlist_layout_no_results) lateinit var noResultsView: View
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_watchlist)
+        ButterKnife.bind(this)
+
+        setSupportActionBar(toolbar)
+        supportActionBar!!.setHomeButtonEnabled(true)
+
+        dealsList.setPresenter(dealsListPresenter)
```

Comment:
No - the reason I think that DealListViewImpl was originally created was to share code between DealsFragment and WatchListActivity since they show similar things. There are better ways to do that though so we'll work towards removing it as we go.